packages/langium/test/parser/langium-parser-builder.test.ts

```diff
@@ -259,6 +259,42 @@ describe('Parser calls value converter', () => {
     });
 });
 
+describe('Parsing with Inferred Types', () => {
```

Comment:
It may be better to move these tests to `type-validator.test.ts`, since they are more related to type inference than to the parser. Here you can check existing of errors using this:
`expect(diagnostics.filter(d => d.severity === DiagnosticSeverity.Error)).toHaveLength(2);`

Reply:
So these tests actually are tied into parser generation. Generating a previously problematic grammar produces no diagnostics, but when you try to build a parser it would fail.